geom/type_line_string.go

```diff
@@ -296,3 +296,17 @@ func (s LineString) Length() float64 {
 func (s LineString) AsMultiLineString() MultiLineString {
 	return NewMultiLineString([]LineString{s})
 }
+
+// Reverse in the case of LineString outputs the coordinates in reverse order.
+func (s LineString) Reverse() Geometry {
+	coords := s.Coordinates()
+	// Reverse the slice.
+	for left, right := 0, len(coords)-1; left < right; left, right = left+1, right-1 {
+		coords[left], coords[right] = coords[right], coords[left]
+	}
+	if s, err := NewLineStringC(coords); err != nil {
+		panic("Reverse of an existing LineString should not fail")
+	} else {
```

Comment:
Nit: [Only error flow should be indented](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow), which I would interpret to mean that there shouldn't be an else (and the `return s.AsGeometry()` should go on its own line.

Similar for MultiPolygons and Polygons below.